src/index.js

```diff
@@ -1,30 +1,95 @@
 const express = require("express");
 const bodyParser = require("body-parser");
-
+const passport = require("passport");
+const Strategy = require("passport-local").Strategy;
+const db = require("./db");
 const app = express();
 
+// Configure the local strategy for use by Passport.
+passport.use(
+  new Strategy(function (username, password, cb) {
+    db.users.findByUsername(username, function (err, user) {
+      if (err) {
+        return cb(err);
+      }
+      if (!user) {
+        return cb(null, false);
```

Comment:
Should be `null` so the return type is always an object I think.

Reply:
So I am def not a Javascript person so I could very much be misunderstanding js, but it was my assumption that everything apart from primitives are objects in JS. So as long as we don't return a primitive type it will always be an examinable object.

Reply:
I just meant js lets you return any type from a function, but IMO its easier to reason about if the returned type is always the same. The caller in this case needs to check if it got back an object or boolean in this case.